urlce7759b8da_Palachsuper_EBA4_tgz-pJ8-json_TestJ8/src/terminal/service/Service.java

```diff
@@ -0,0 +1,193 @@
+                     case dispense:
+                         dispencer.main(cmd);
+                         break;
+                     case card:
+                         card_reader.main(cmd);
+                         break;
+                     case cashe:
+                         cashe_machine.main(json_cmd);
+                         break;
+                 }
+             } catch (IllegalArgumentException iae) {
+
+             }
+             String pr_string = json_cmd.getString("params");
+             JSONObject param = new JSONObject(pr_string);
+         }
+         catch (Exception e){
+
+         }
+    }
+
+    public static void WritePaymentLog(String log, String id_payment){
+        StringBuilder sb = new StringBuilder();
+        try{
+            String path_log = "/home/terminal/Log/LogPayment/Payment_"+id_payment+".txt";
+            File Log_path = new File(path_log);
+            if (!Log_path.exists()){
+                try {
+                    Log_path.createNewFile();
+                } catch (IOException e) {
+                    // TODO Auto-generated catch block
+                    e.printStackTrace();
+                }
+            }
+            BufferedReader in = new BufferedReader(new FileReader( new File(path_log).getAbsoluteFile()));
+            String read_log;
+            if (sb.length()>0)
+            {
+                sb.delete(0,sb.length()-1);
+            }
+            while ((read_log = in.readLine()) != null)
+            {
+                if (sb.length()>0)
+                {
+                    sb.append(read_log);
+                    sb.append("\n");
+                }
+            }
+            in.close();
+            String str="";
+            str =sb + log;
+            PrintStream printStream = new PrintStream(new FileOutputStream(path_log, true), true);
+            printStream.println(str);
+            printStream.close();
+        }
+        catch (Exception e){
+
+        }
+    }
+
+    public static void WriteLog(String log){
+        String dat_time_path="";
+        StringBuilder sb = new StringBuilder();
+        try
+        {
+            try
+            {
+                Calendar cal = new GregorianCalendar();
+                String name_cal=new SimpleDateFormat("yyyyMMdd").format(cal.getTime());
+                dat_time_path= "/home/terminal/Log/LogSystem/log"+name_cal+".txt";
+                //dat_time_path= "/home/terminal/Log/print_query.txt";
+                File Log_path = new File(dat_time_path);
+                if (!Log_path.exists())
+                {
+                    try {
+                        Log_path.createNewFile();
+                    } catch (IOException e) {
+                        // TODO Auto-generated catch block
+                        e.printStackTrace();
+                    }
+                }
+                BufferedReader in = new BufferedReader(new FileReader( new File(dat_time_path).getAbsoluteFile()));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *in*. The resource is referenced at lines 160, 166. The resource is closed at line 174. There are other execution paths that do not contain closure statements, for example, when BufferedReader.readLine throws an exception. Ensure *in* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).